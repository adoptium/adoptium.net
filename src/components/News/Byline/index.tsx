import React from "react"
import Link from "next/link"
import ProfilePicInline from "@/components/ProfilePicInline"

interface IBylineProps {
  author: string
  date: string
  identifier: string
}

const Byline: React.FC<IBylineProps> = props => {
  const { author, date, identifier } = props
  return (
    <div className="flex justify-center items-center gap-5">
      <ProfilePicInline identifier={identifier} name={author} />
      <span className="text-[16px] font-bold leading-[150%] text-white">
        <Link href={`/news/author/${identifier}`}>
          <span className="text-primary pr-2">{author},</span> {date}
        </Link>
      </span>
    </div>
  )
}

export default Byline
