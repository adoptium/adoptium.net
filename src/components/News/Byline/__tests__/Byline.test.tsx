import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import Byline from "../index"

describe("Byline component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Byline author="Adoptium PMC" date="2021-07-01" identifier="pmc" />,
    )

    expect(container).toMatchSnapshot()
  })
})
