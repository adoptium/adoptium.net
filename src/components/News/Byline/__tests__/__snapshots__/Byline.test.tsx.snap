// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Byline component > should render correctly 1`] = `
<div>
  <div
    class="flex justify-center items-center gap-5"
  >
    <img
      alt="Adoptium PMC"
      image="[object Object]"
      stlye="[object Object]"
      style="margin-left: 0.5rem; margin-bottom: 0px; min-width: 30px; border-radius: 100%;"
    />
    <span
      class="text-[16px] font-bold leading-[150%] text-white"
    >
      <a
        href="/en/news/author/pmc"
      >
        <span
          class="text-primary pr-2"
        >
          Adoptium PMC
          ,
        </span>
         
        2021-07-01
      </a>
    </span>
  </div>
</div>
`;
